### ISSUE TITLE
How to get database working with scrapy pipeline

### ISSUE THREAD

**Issue by GaelGil** [GaelGil/web-scraper#1]
I have this on a separate directory working. The data gets added manually
every time.
~~~python
DeclarativeBase = declarative_base()

def db_connect():
    return create_engine('sqlite:///database.db')

def create_table(engine):
    DeclarativeBase.metadata.create_all(engine)

class SneakerDB(DeclarativeBase):
    __tablename__ = "sneaker_table"

    id = Column(Integer, primary_key=True)
    sneaker = Column('sneaker', Text())
    price = Column('price', Integer())

engine = db_connect()
create_table(engine)
Session = sessionmaker(bind=engine)
session = Session()
sneakers = SneakerDB()
sneakers.sneaker = "sneaker"
sneakers.price = 1000

try:
    session.add(sneakers)
    session.commit()
except:
    session.rollback()
    raise
finally:
    session.close()
~~~

In this repo I have the creating of the database and table separate in`models.py`.
~~~python
DeclarativeBase = declarative_base()

def db_connect():
    """
    start the database
    """
    return create_engine('sqlite:///database.db')

def create_table(engine):
    DeclarativeBase.metadata.create_all(engine)

class SneakerDB(DeclarativeBase):
    """
    create the table and somecolumns
    """
    __tablename__ = "sneaker_table"

    id = Column(Integer, primary_key=True)
    sneaker = Column('sneaker', Text())
    price = Column('price', Integer())

~~~

Then to store it I have this`pipelines.py`.
~~~python
class ShoesPipeline(object):
    def __init__(self):
        """
        starts database connection and creates table, 
        as well as session
        """
        engine = db_connect()
        create_table(engine)
        self.Session = sessionmaker(bind=engine)

    def process_item(self, item, spider):
        """
        """
        session = self.Session()
        sneakers = SneakerDB()
        sneakers.sneaker = item['shoe'],
        sneakers.price = item['price']

        try:
            session.add(sneakers)
            session.commit()
        except:
            session.rollback()
            raise
        finally:
            session.close()

        return item

~~~

In the end the database gets created but it's not being populated with the data, I don't really know what the issue is and what to do. And I get this error
`
sqlalchemy.exc.InterfaceError: (sqlite3.InterfaceError) Error binding parameter 0 - probably unsupported type.
[SQL: INSERT INTO sneaker_table (sneaker, price) VALUES (?, ?)]
[parameters: (['adidas Yeezy Boost 700 V2 Geode'], ['$259'])]
(Background on this error at: http://sqlalche.me/e/rvf5)
`

**Comment by camoverride** [comment]
I have two pieces of advice:

1. read through every line in the code and make sure you understand what everything actually does.  For instance, can you figure out what `DeclarativeBase = declarative_base()` is actually doing? If you go through that exercise, it will make all the following debugging easier.

2. It looks like you have a type error.

~~~python
    sneaker = Column('sneaker', Text()) # declares that the value should be a string, str
    price = Column('price', Integer())      # declares that the value should be an integer, int
~~~

The values you are trying to add both look like strings:
`['adidas Yeezy Boost 700 V2 Geode'], ['$259']`

Test it for yourself in the python interpreter:
~~~shell
>>> type("$259")
<class 'str'>
~~~

**Comment by camoverride** [comment]
One more thing to keep in mind: python is very generous with the kind of things you can do with different types of data:

~~~python
list = [1, [2,3], "string"] # a list containing different types
~~~

Most languages don't allow this. SQL only lets one kind of data belong in each column: `Text()` or `Integer()` which are presumably like `str` and `int` in Python.

**Comment by GaelGil** [comment]
Fixed in Latest commit `821d16a`. Previously I had
~~~python     
sneakers.price = item['price']
~~~

Now this. 

~~~python
        #this converts '$219' into 219.
        sneakers.price = int(item['price'][0][1:])
~~~
